File: lib/domain/entities/job.dart
/// Job Intelligent - Job Entity
/// Domain entity representing a job posting.
library;

class JobEntity {
  final int id;
  final String title;
  final String description;
  final String company;
  final String location;
  final String jobType;
  final List<String> skillsRequired;
  final String source;
  final DateTime? createdAt;

  const JobEntity({
    required this.id,
    required this.title,
    required this.description,
    required this.company,
    required this.location,
    required this.jobType,
    required this.skillsRequired,
    required this.source,
    this.createdAt,
  });
}

File: backend/models/job.py
"""
Job Intelligent - Job Model
SQLAlchemy model for the jobs table.
"""
from sqlalchemy import Column, Integer, String, Text, DateTime
from sqlalchemy.dialects.postgresql import ARRAY
from sqlalchemy.sql import func

from backend.database.session import Base


class Job(Base):
    """Job posting model."""
    __tablename__ = "jobs"

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String(255), nullable=False, index=True)
    description = Column(Text, nullable=False)
    company = Column(String(255), nullable=False)
    location = Column(String(255), nullable=False, index=True)
    job_type = Column(String(50), default="CDI")  # CDI, CDD, Stage, Freelance
    skills_required = Column(ARRAY(String), default=[])
    source = Column(String(100), default="internal")
    created_at = Column(DateTime(timezone=True), server_default=func.now())
